app/controllers/concerns/api_helper.rb

```diff
@@ -36,9 +36,8 @@ def verify_checksum(force_loadbalancer_secret = false)
 
     # Camel case (ex) get_meetings to getMeetings to match BBB server
     check_string = action_name.camelcase(:lower)
-    check_string += request.query_string.gsub(
-      /&checksum=#{params[:checksum]}|checksum=#{params[:checksum]}&|checksum=#{params[:checksum]}/, ''
-    )
+    # Generate a new query string using only allowed params (cant use .to_query because it changes the order)
+    check_string += request.query_parameters.except(:checksum).map { |k, v| "#{CGI.escape(k.to_s)}=#{CGI.escape(v)}" }.join('&')
```

Comment:
This doesn't fix the problem. The `request.query_parameters` method still only returns query parameters, not parameters in a post body.
The use of the `CGI.escape` function might escape the parameters differently from how they were on the original request (for example, integrations may either use `%20` vs `+` to replace a space), which will cause the checksum to fail in some cases.

Reply:
`query_parameters` reflects the same parameters that are passed to `encode_bbb_uri` through this method here

```
def pass_through_params(excluded_params)
    params.except(*(excluded_params + [:format, :controller, :action, :checksum]))
          .to_unsafe_hash
  end
```

It doesn't fully include the body but at the very least it makes the checksum check the same parameters that are passed through to BBB